packages/react-components/src/components/PromoBanner/PromoBanner.tsx

```diff
@@ -32,23 +30,42 @@ export const PromoBanner: React.FC<PromoBannerProps> = ({
   img,
   light = false,
   linkText,
-  size = 'small',
   onButtonClick,
   onClose,
   onLinkClick,
 }) => {
+  const containerRef = React.useRef<HTMLDivElement>(null);
+  const [isSmallContainer, setIsSmallContainer] = React.useState(false);
+  const [isLargeContainer, setIsLargeContainer] = React.useState(false);
+
   const mergedClassNames = cx(
     styles[baseClass],
     {
       [styles[`${baseClass}--light`]]: light,
-      [styles[`${baseClass}--${size}`]]: size,
+      [styles[`${baseClass}--small`]]: isSmallContainer,
+      [styles[`${baseClass}--large`]]: isLargeContainer,
     },
     className
   );
 
-  const shouldRenderLargeFooter = (buttonText || linkText) && size === 'large';
-  const shouldRenderSmallOrMediumFooter =
-    (buttonText || linkText) && size !== 'large';
+  React.useEffect(() => {
+    const handleResize = debounce(() => {
+      if (containerRef.current && containerRef.current.offsetWidth <= 400) {
```

Comment:
please add those 400 and 800 thresholds to the const above or maybe to the upper level if those are reusable across other components